Sofia Lennbom Portfolio

solen80a opened the pull request:
Please include a link to your Figma design and a Netlify link.

https://www.figma.com/design/RsWkRToMb9XaYPNpNNDdo6/Sofia%C2%B4s-Figma-designs-for-students?t=C4rBuD1OB9kyxHsW-0
https://sofialennbom.netlify.app/

I ran out of time and some stuff are still not perfect.
No Nav bar is added and tablet is not perfect.
And I did not manage to get the project cards to reverse odd/even.
Probably some other stuff that I have missed.

solen80a committed:
Added a header with navigation.

solen80a committed:
Rearranged so main.jsx only contains the original stuff. Devided project into two files.

solen80a committed:
Rearranged ProjCards.

solen80a committed:
Implemented a hamburger menu with userState.

solen80a committed:
Hamburger is positioned correct and only show on mobile. Naviagtion for desktop got background color. Profile img is showing.

solen80a committed:
Added footer and profile component with styling.

solen80a committed:
Updated proj with project content from app. Added tags and netlify link.

solen80a committed:
Added skip to main content.

solen80a committed:
Hamburger opens, and fixed open x.

solen80a committed:
Added svg icons and imgs. Added styled components. Added a styled button, not finished.

solen80a committed:
Moved css styling to styled components for Profile, Tech and Footer.

solen80a committed:
Added skip link as it´s own section component with styled components.

solen80a committed:
Add MyWordsCard component and MyWords section with data integration

- Created MyWordsCard component to display individual articles with image, title, content, and a button.
- Added mywords.json file containing an array of articles with relevant data.
- Implemented MyWords section to map through the articles and render MyWordsCard for each item.
- Updated Skills section with basic structure for future content.

solen80a committed:
Add projects data integration and update components to utilize project URLs

solen80a committed:
Refactor components and styles for improved structure and readability; update project tags to use buttons in Proj section.

solen80a committed:
Refactor Proj and Skills sections to use TagButton component for tags; enhance Skills section layout with styled components.

solen80a committed:
Refactor Proj section to utilize TagButton component for tags; remove unused CSS files and adjust button styles for better responsiveness.

solen80a committed:
Update README and Footer components; enhance project details and add social media links

solen80a committed:
Implement code changes to enhance functionality and improve performance

solen80a committed:
Add MyWordsCard component and MyWords section with data integration

- Created MyWordsCard component to display individual articles with image, title, content, and a button.
- Added mywords.json file containing an array of articles with relevant data.
- Implemented MyWords section to map through the articles and render MyWordsCard for each item.
- Updated Skills section with basic structure for future content.

solen80a committed:
Refactor components for improved styling and responsiveness; update Button, GlobalStyle, MyWordsCard, ProjCard, Footer, and Profile sections.

solen80a committed:
Refactor media queries for improved responsiveness; update GlobalStyle, Media, ProjCard, and Profile components.

solen80a committed:
Refactor code structure for improved readability and maintainability

solen80a committed:
Refactor GlobalStyle, Profile, Skills, and Tech components for improved responsiveness and layout consistency

solen80a committed:
Refactor Button and ProjCard components for improved layout and responsiveness

solen80a committed:
Refactor GlobalStyle, MyWordsCard, and Profile components for improved layout, responsiveness, and semantic structure

solen80a committed:
Refactor GlobalStyle and MyWordsCard components for improved responsiveness and layout consistency

solen80a committed:
Refactor MyWordsCard and ProjCard components for improved semantic structure and layout consistency

solen80a committed:
Cleaned code and removed unused files.

HIPPIEKICK reviewed (changes requested):
**A11y**
100% in Lighthouse ⭐

**Responsiveness**
- Requirement not met: Your portfolio should be responsive (it should look good on devices from 320px width up to at least 1600px)

**Regarding the requirement: Your portfolio should follow the Figma design**
- Some sections need a max-width
- Make sure you have the right top/bottom padding for the sections
- Project section look a little chaotic. Look over the margin/padding as well as the positioning of tags vs image
- Check so that all elements have the correct font (e.g. Skills’ headings don’t)

Overall it looks good, but there are some additional tweaks needed for it to be following the design closely.

**Codewise**
Good job structuring your project and making use of components and props 👍 You kept your entry point (App.jsx) clean which makes the project easy to overview.

**Changes requested**
- Responsiveness
- Have a look at the comments I wrote about the design

VariaSlu commented:
Hi Sofia,  
You’ve done a great job — your portfolio already looks like something you could proudly send to a future employer or client.

I believe you’ve done the best you could in the given timeframe. Your website presents a clean and personal online portfolio. It’s visually minimal, easy to navigate, and tailored for showcasing your work and personality.

Accessibility is over the moon! Maybe it would be even 150% in Lighthouse if checking for meta tags and alt text.

Consider including more alt text for images for better accessibility.

Your project has a solid, well-organized file structure. Clean and component-based.

I especially like that:
Logic, data, and presentation are separated
Reusable components like TagButton and Typography help maintain clean code
File and folder names are meaningful and consistent

Consider grouping GlobalStyle.jsx, Media.jsx, and Typography.jsx into a styles/ folder to better organize styling logic.
I wouldn’t mind if you renamed the component ProjCard to ProjectCard, but that’s totally up to you since it is a personal project :)

Styling suggestion for this component: ProjectCardContent is missing display: flex or flex-direction: column, which might be required for gap to take effect in some browsers.

Thank you for your patience, and good luck with improving this one further or building a whole new portfolio in the future!

solen80a committed:
Fixed padding and max-width on sections. Also fixed the font on tags.

solen80a committed:
Added even odd to my project card and mywords cards. Also added padding to footer.

solen80a committed:
Made some cleaning and fixed profile img.

solen80a committed:
Fixed styling for sections and img.

solen80a committed:
Made some more cleaning.

solen80a commented:
Hi Matlida,
Fixed styling on sections and the font on tags. Added so the project card and the My words card have the even/odd. Made some more cleaning of unused code. I think the responsiveness look good now.
Ready for a new review!

JennieDalgren commented:
@solen80a Hi Sofia! Take a look at the padding/margin on some sections. The text and other elements shouldnt go all the way to the edge, it looks a bit cramped. Screenshot for example: 
<img width="941" alt="image" src="https://github.com/user-attachments/assets/3a2382e1-3040-4893-9abd-41338c7bce76" />
<img width="891" alt="image" src="https://github.com/user-attachments/assets/60807b4c-cce3-430f-b990-f8367dfea16e" />

solen80a committed:
Added mor padding to the sections. And some more styling.

solen80a commented:
Hi Jennie,
Missed that, added som more padding to the sections.
Please review.

HIPPIEKICK commented:
Some changes still needed:
- Still not responsive, have a look at mobile (e.g. 320px width)
- Look over all spacing in the design and implement in your portfolio. Consistency is key 🚀

solen80a committed:
Styling changes.

solen80a committed:
More styling fixes to get everything fit on mobile. Also made som cleaning of unused code.

solen80a commented:
Hi, made some fixes, please review.

HIPPIEKICK reviewed (approved):
⭐

solen80a committed:
Added project to portfolio and updated skills list.
